src/Umbraco.Web.Common/Views/UmbracoViewPage.cs

```diff
@@ -9,250 +8,254 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Options;
 using Umbraco.Cms.Core.Configuration.Models;
-using Umbraco.Cms.Core.IO;
 using Umbraco.Cms.Core.Logging;
 using Umbraco.Cms.Core.Models;
 using Umbraco.Cms.Core.Models.PublishedContent;
 using Umbraco.Cms.Core.Strings;
 using Umbraco.Cms.Core.Web;
 using Umbraco.Cms.Web.Common.ModelBinders;
 using Umbraco.Extensions;
+using IHostingEnvironment = Umbraco.Cms.Core.Hosting.IHostingEnvironment;
 
-namespace Umbraco.Cms.Web.Common.Views
+namespace Umbraco.Cms.Web.Common.Views;
+
+public abstract class UmbracoViewPage : UmbracoViewPage<IPublishedContent>
 {
-    public abstract class UmbracoViewPage : UmbracoViewPage<IPublishedContent>
-    {
+}
 
-    }
+public abstract class UmbracoViewPage<TModel> : RazorPage<TModel>
+{
+    private UmbracoHelper? _helper;
 
-    public abstract class UmbracoViewPage<TModel> : RazorPage<TModel>
+    /// <summary>
+    ///     Gets the Umbraco helper.
+    /// </summary>
+    public UmbracoHelper Umbraco
     {
-        private IUmbracoContext? _umbracoContext;
-        private UmbracoHelper? _helper;
+        get
+        {
+            if (_helper != null)
+            {
+                return _helper;
+            }
 
-        private IUmbracoContextAccessor UmbracoContextAccessor => Context.RequestServices.GetRequiredService<IUmbracoContextAccessor>();
+            TModel model = ViewData.Model;
+            var content = model as IPublishedContent;
+            if (content is null && model is IContentModel contentModel)
+            {
+                content = contentModel.Content;
+            }
 
-        private GlobalSettings GlobalSettings => Context.RequestServices.GetRequiredService<IOptions<GlobalSettings>>().Value;
+            if (content is null)
+            {
+                content = UmbracoContext?.PublishedRequest?.PublishedContent;
+            }
 
-        private ContentSettings ContentSettings => Context.RequestServices.GetRequiredService<IOptions<ContentSettings>>().Value;
+            _helper = Context.RequestServices.GetRequiredService<UmbracoHelper>();
 
-        private IProfilerHtml ProfilerHtml => Context.RequestServices.GetRequiredService<IProfilerHtml>();
+            if (!(content is null))
+            {
+                _helper.AssignedContentItem = content;
+            }
```

Comment:
@Zeegaan couldn't this part be simplified to:

```
TModel model = ViewData.Model;
var content = model as IPublishedContent;
if (content is null && model is IContentModel contentModel)
{
    content = contentModel.Content;
}

_helper = Context.RequestServices.GetRequiredService<UmbracoHelper>();

content ??= UmbracoContext?.PublishedRequest?.PublishedContent;

if (content is not null)
{
    _helper.AssignedContentItem = content;
}
```

Reply:
Hey @bjarnef you are totally right that could be simplified, could i trouble you to make a PR with the changes? I will gladly review it 😄

Reply:
@Zeegaan here you go 😁  https://github.com/umbraco/Umbraco-CMS/pull/12528